compositorclient: add SurfaceByName() and NativeSurface() apis

petegriffin opened the pull request:
Re-worked previous implementation a bit following Pierres review
feedback to better map to the existing IDisplay & ISurface classes.

SurfaceByName() is used on the IDisplay class to get the ISurface.
NativeSurface() is used on the ISurface to get the native surface.

These api's are used with Wayland Weston compositor client when
using waylandsink as the video sink in webkit. I've moved the implementation
to the Implementation.h header which is shared between Westeros and Weston.

wpebackend-rdk calls these apis to obtain the surface handle created by
Thunder, which is then passed to waylandsink as an external surface so
that the video surface is correctly created as a sub-surface
of the browser.

Also RPI compositorclient is updated to have a stub implementation.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>

petegriffin commented:
@pwielders Hi Pierre, I've re-worked this I think to what you had in mind in your review feedback over email.

I've added back the api to Client.h again, but this time I've provided a Westeros and RPI stub implementation. So hopefully this no longer breaks your build.

I've also updated wpebackend-rdk to use these 2 new APIs.

pwielders commented:
Yip this is indeed more in line with how the navigation runs through the Client interfaces, however, i think that on ISurface, the method Native and NativeSurface return the same, so I would expect the new method is not needed here??? Not [sure](url) but if I understand correctly, the  EGLNativeWindowType is a mapping to void* as well..

petegriffin commented:
@pwielders Hi Pierre, I looked into your suggestion, and EGLNativeWindow maps to wl_egl_window on wayland which contains the wl_surface which is what I need.

So in v3 I've removed the NativeSurface() api and updated wpebackend-rdk to extract the wl_surface from the EGLNativeWindow handle returned by existing Native() api. You can see the updated corresponding wpebackend-rdk code here https://github.com/WebPlatformForEmbedded/WPEBackend-rdk/pull/55

pwielders commented:
Looks good now. Thanks @petegriffin for the contribution !
